### PR TITLE
fix: fix StringRef pointer in func ASTIntLiteralToNum

### LINKED ISSUE #516: Bug: ast_node_converter_test run failed
OpenMLDB/hybridse/src/planv2/ast_node_converter_test.cc:861: Failure
Expected equality of these values:
  s2.code
    Which is: 1002
  common::kOk
    Which is: 0
[  FAILED  ] ASTNodeConverterTest.ASTIntLiteralToNumberTest (0 ms)

### PR DESCRIPTION
**Pull request by zoyopei**
* **Please check if the PR fulfills these requirements**

- The commit message follows our guidelines
- Tests for the changes have been added (for bug fixes / features)
- Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Issue: #516 

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: hybridse/src/planv2/ast_node_converter.cc [hunk 1]
```diff
@@ -1569,7 +1569,7 @@ base::Status ASTIntLiteralToNum(const zetasql::ASTExpression* ast_expr, int64_t*
     bool is_null = false;
     if (int_literal->is_long()) {
         const int size = int_literal->image().size();
-        codec::StringRef str_ref(std::string(int_literal->image().substr(0, size - 1)));
+        codec::StringRef str_ref(size - 1, int_literal->image().data());
```
Comment (aceforeverd) [inline hybridse/src/planv2/ast_node_converter.cc:1642]: what's the difference ?

Comment (zoyopei) [reply hybridse/src/planv2/ast_node_converter.cc:1642]: std::string constructs a temporary variable, it will be destructed immediately

Comment (jingchen2222) [reply hybridse/src/planv2/ast_node_converter.cc:1642]: If this is a problem, we should fix other lines as well.  @aceforeverd 
@zoyopei would you mind helping us fix other lines in ast_node_convert.cc
e.g., 
```c++
codec::StringRef str_ref(std::string(int_literal->image()));
        udf::v1::string_to_bigint(&str_ref, val, &is_null);
```

Comment (aceforeverd) [reply hybridse/src/planv2/ast_node_converter.cc:1642]: damn right

**Comment by zoyopei** [comment]
new RP #551

**Pull request closed without merging**